### DISCUSSION TITLE
Add client-side cvar that allows changing textures on maps

### DISCUSSION THREAD

**Discussion by ETLIssues** [etlegacy/etlegacy#2309] [Ideas]
I'll start this by acknowledging my idea is almost certainly not feasible. Nonetheless, I'd really love to be able to change the environment on certain maps. Fueldump is a very fun map, but after a 30-minute round, my eyes are very tired from the overwhelming white. Further, when it's been snowing at my home for 3 months, I'd really like to look at some grass in the game. It'd also just make old maps seem fresh.

I can imagine it being an enum cvar where you toggle from snow to grass to desert etc., or I can imagine a str cvar where you enter a texture path, or a separate config file outside the game.

I personally feel that there is no competitive advantage in, for example, changing frostbite to desert, but it's important to note that servers and leagues could restrict players' ability to use this feature.

**Comment by oheikk** [comment]
There's a simpler way to solve this problem, if the admins of the servers you're playing on are open to it.

Servers can provide optionally downloadable texture packs that override whatever textures you want.  Hirntot had that feature at least back then I played there, maybe they still do. @x0rnn can probably better provide details but it should be fairly simple for any server admin to set up.

The client-side feature is IMO not worth it, and there are potential exploit issues that might be hard to deal with (what about transparent/invisible textures, making player models bright and pink etc.)

**Reply by ETLIssues** [reply]
> Servers can provide optionally downloadable texture packs that override whatever textures you want. Hirntot had that feature at least back then I played there, maybe they still do.

Are you saying that players would be able to choose a snow version, a desert version, a grass version, etc. while all playing on the same server with each other? Very cool if so and does address my problem.

> The client-side feature is IMO not worth it, and there are potential exploit issues that might be hard to deal with (what about transparent/invisible textures, making player models bright and pink etc.)

I did consider invisible textures, that's why I would lean toward the enum cvar where it's just a preset list of textures that can be selected. However, I am uncertain where the texture paths for that cvar would be stored, so it's definitely possible that it wouldn't address the abuse.

**Reply by oheikk** [reply]
> Are you saying that players would be able to choose a snow version, a desert version, a grass version, etc. while all playing on the same server with each other?

Yes. Admins need to add files such as `z_fueldump_summer.pk3` in their server's etmain folder, that have texture overrides in them. They can then provide download links and instructions for players to put the file in their etmain e.g. in server banners, Discord, forums etc. It's a bit of work to create them, but Hirntot to the rescue: https://hirntot.org/viewtopic.php?f=20&t=47

**Comment by x0rnn** [comment]
Yea, server admins can put optional pk3s inside the etmain folder. We have an optional spring texture pack for fueldump and railgun and a snow version of radar on Hirntot.

**Comment by casaluigi2014-a11y** [comment]
My server offers over 20 maps with restored textures in hdr mode with very high clarity. UltraX Mod is another mod in testing. search for the server and try both the maps and the mod.
